modules/pricer/src/main/java/com/opengamma/strata/pricer/fx/SmileDeltaTermStructureParameters.java

```diff
@@ -5,479 +5,202 @@
-        volatilitySensitivity[looptime][loopvol] = volatilitySensitivityVol[looptime] * volatilityAtTimeSensitivity[loopvol];
+        volatilitySensitivity[looptime][loopvol] =
+            volatilitySensitivityVol[looptime] * volatilityAtTimeSensitivity.get(loopvol);
       }
-      volatilityT[loopvol] = _timeInterpolator.interpolate(interpData, time);
+      volatilityT[loopvol] = bound.interpolate(expiry);
     }
-    SmileDeltaParameters smile = SmileDeltaParameters.of(time, _volatilityTerm[0].getDelta(), DoubleArray.copyOf(volatilityT));
+    SmileDeltaParameters smile = SmileDeltaParameters.of(expiry, getDelta(), DoubleArray.copyOf(volatilityT));
     return SmileAndBucketedSensitivities.of(smile, DoubleMatrix.copyOf(volatilitySensitivity));
   }
 
   /**
-   * Gets the name of smile parameter term structure.
+   * Gets the number of smiles.
    * 
-   * @return the name
+   * @return the number of smiles
    */
-  public String getName() {
-    return _name;
+  default int getNumberSmiles() {
+    return getVolatilityTerm().size();
   }
 
   /**
-   * Gets the times to expiry.
-   * @return The times.
+   * Gets the number of strikes. 
+   * 
+   * @return the number of strikes
    */
-  public double[] getTimeToExpiry() {
-    return _timeToExpiry;
+  default int getNumberStrike() {
+    return getVolatilityTerm().get(0).getVolatility().size();
   }
 
   /**
-   * Gets the number of expirys.
-   * @return The number of expirys.
+   * Gets delta values.
+   * 
+   * @return the delta values
    */
-  public int getNumberExpiry() {
-    return _timeToExpiry.length;
+  default DoubleArray getDelta() {
+    return getVolatilityTerm().get(0).getDelta();
   }
 
   /**
-   * Gets the time interpolator
-   * @return The time interpolator
+   * Gets a set of expiry for smiles.
+   * 
+   * @return the set of expiry
    */
-  public Interpolator1D getTimeInterpolator() {
-    return _timeInterpolator;
-  }
+  public DoubleArray getTimeToExpiry();
 
   /**
-   * Gets the volatility smiles from delta.
-   * @return The volatility smiles.
+   * Gets the time left extrapolator. 
+   * 
+   * @return the time left extrapolator
    */
-  public SmileDeltaParameters[] getVolatilityTerm() {
-    return _volatilityTerm;
-  }
+  public CurveExtrapolator getTimeLeftExtrapolator();
 
   /**
-   * Gets the number of strikes (common to all dates).
-   * @return The number of strikes.
+   * Gets the time interpolator. 
+   * 
+   * @return the time interpolator
    */
-  public int getNumberStrike() {
-    return _volatilityTerm[0].getVolatility().size();
-  }
+  public CurveInterpolator getTimeInterpolator();
 
   /**
-   * Gets delta (common to all time to expiry).
-   * @return The delta.
+   * Gets the time right extrapolator. 
+   * 
+   * @return the time right extrapolator
    */
-  public double[] getDelta() {
-    return _volatilityTerm[0].getDelta().toArray();
-  }
+  public CurveExtrapolator getTimeRightExtrapolator();
```

Comment:
Exposing the interpolators/extrapolators feels too specific for a top level merket data object. Seems like the methods that rely on them should move down into the implementation to keep the interface "clean"